sys/include/entropy_source/adc_noise.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup   sys_entropy_source_adc ADC Noise Entropy Source
+ * @ingroup    sys_entropy_source
+ * @brief      Entropy Source based on LSB of ADC samples
+ *
+ * This module provides entropy from ADC samples. Thereby, only the LSB is considered for
+ * generation of output values. The available resolution as well as internal sampling rates,
+ * specific driver settings and device inaccuracies may lead to different behavior between
+ * different platforms. The configured ADC pin might be unconnected and floating,
+ * it can be exposed as an I/O pin or internally connected to a thermal noise source, an
+ * avalanche diode circuit or the receive path of an antenna. The possibilities are manifold,
+ * thus, configuration and deployment properties of this module need to be thoroughly
+ * validated.
+ *
+ * @note It is worth noting that ADC pins are typically vulnerable when physically
+ *       accessible. Developers should consider additional tamper detection concepts
+ *       and enclosures.
+ *
+ * @{
+ * @file
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ */
+
+#ifndef ENTROPY_SOURCE_ADC_NOISE_H
+#define ENTROPY_SOURCE_ADC_NOISE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <inttypes.h>
+#include <stddef.h>
+
+#include "periph/adc.h"
+
+/**
+ * @ingroup    sys_entropy_source_config
+ * @{
+ */
+/**
+ * @brief   ADC resolution default configuration.
+ */
+#if IS_ACTIVE(CONFIG_ENTROPY_SOURCE_ADC_RES_6BIT)
+#define CONFIG_ENTROPY_SOURCE_ADC_RES          ADC_RES_6BIT
+#else if IS_ACTIVE(CONFIG_ENTROPY_SOURCE_ADC_RES_8BIT)
```

Comment:
Here and below:

```suggestion
#elif IS_ACTIVE(CONFIG_ENTROPY_SOURCE_ADC_RES_8BIT)
```

;-)